### ISSUE TITLE
TAblePlus keeps prompting for ssh password even tho save in keychain is selected

### ISSUE THREAD

**Issue by adamwinn** [TablePlus/TablePlus#1682]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):

3. The steps to reproduce this issue:

Noted: If the bug is related to data, please attach an example SQL data.

**Comment by huyphams** [comment]
Thanks @adamwinn, please turn on SSH debug log and see what was happening, you can  turn it on via the main menu:
<img width="486" alt="Screen Shot 2020-01-25 at 7 28 15 AM" src="https://user-images.githubusercontent.com/3654622/73120434-44a1c580-3f44-11ea-9178-956776ce9ff3.png">

**Comment by adamwinn** [comment]
It says towards the end:
```
[2020/01/25 04:40:53.574289, 1] ssh_packet_userauth_failure:  Access denied for 'password'. Authentication that can continue: publickey,keyboard-interactive
[2020/01/25 04:40:53.574301, 2] ssh_packet_userauth_failure:  Access denied for 'password'. Authentication that can continue: publickey,keyboard-interactive
[2020/01/25 04:40:53.574318, 3] ssh_userauth_kbdint_init:  Sending keyboard-interactive init request
[2020/01/25 04:40:53.574356, 3] ssh_socket_unbuffered_write:  Enabling POLLOUT for socket
[2020/01/25 04:40:53.574365, 3] packet_send2:  packet: wrote [type=50, len=76, padding_size=15, comp=60, payload=60]
[2020/01/25 04:40:53.574381, 4] ssh_socket_pollcallback:  Poll callback on socket 11 (POLLOUT ), out buffer 0
[2020/01/25 04:40:53.574389, 4] ssh_socket_pollcallback:  sending control flow event
[2020/01/25 04:40:53.574394, 4] ssh_packet_socket_controlflow_callback:  sending channel_write_wontblock callback
[2020/01/25 04:40:53.655421, 4] ssh_socket_pollcallback:  Poll callback on socket 11 (POLLIN ), out buffer 0
[2020/01/25 04:40:53.655462, 3] ssh_packet_socket_callback:  packet: read type 60 [len=44,padding=11,comp=32,payload=32]
[2020/01/25 04:40:53.655470, 3] ssh_packet_process:  Dispatching handler for packet type 60
[2020/01/25 04:40:53.655476, 4] ssh_packet_userauth_pk_ok:  Received SSH_USERAUTH_PK_OK/INFO_REQUEST/GSSAPI_RESPONSE
[2020/01/25 04:40:53.655481, 4] ssh_packet_userauth_pk_ok:  keyboard-interactive context, assuming SSH_USERAUTH_INFO_REQUEST
[2020/01/25 04:40:53.655490, 3] ssh_packet_userauth_info_request:  1 keyboard-interactive prompts
```

**Comment by huyphams** [comment]
Hi @adamwinn sorry because of the late response. As far as I can see that wasn't SSH private key password popup. It was something else that requested by the keyboard-interactive. Could you double-check the private key again?

**Comment by adamwinn** [comment]
I'm not using a private key.  The prompt happens and once I input the password for the ssh server, I'm able to get in.  Using the Test button, it tests successfully with no prompt, but once you hit Connect that prompt happens.  I was expecting my password to be saved in the keychain and then used automatically so I wouldn't have to enter it again

**Comment by huyphams** [comment]
TablePlus will remember the password, I think there are something wrong with the config, please make sure that you enter the correct password. As I mentioned above TablePlus prompts because of the the keyboard interactive 
`[2020/01/25 04:40:53.655490, 3] ssh_packet_userauth_info_request:  1 keyboard-interactive prompts`

**Comment by adamwinn** [comment]
If I didn't enter the password correctly, it wouldn't Test successfully

**Comment by huyphams** [comment]
I mean, the `keyboard-interactive prompts` are some questions that were asked during authenticating. TablePlus couldn't understand the questions. All it does is forwarding the question/answer between the user and the server.

**Comment by CalumTowers** [comment]
I appear to be getting this issue on the iOS app, but not on the desktop app - same server / key. Is it worth a separate bug report in the iOS repo or is there anything I can add to this issue to help (e.g. debugging info)?

**Comment by adamwinn** [comment]
For what it's worth, SequelPro doesn't prompt with the exact same setup

**Comment by Ignas2526** [comment]
I'm having same issue. As far as I understand, problem is that TablePlus does not emulate keyboard-interactive authentication method. Can't you just send key codes or something like that to SSH?
```
[2020/04/19 00:01:17.080111, 1] ssh_packet_userauth_failure:  Access denied for 'password'. Authentication that can continue: publickey,keyboard-interactive
...
[2020/04/19 00:01:17.286589, 4] ssh_packet_userauth_pk_ok:  keyboard-interactive context, assuming SSH_USERAUTH_INFO_REQUEST
[2020/04/19 00:01:17.286603, 3] ssh_packet_userauth_info_request:  1 keyboard-interactive prompts
```

**Comment by huyphams** [comment]
Your log indicates that the pass was asked by keyboard-interactive mode, please double check your ssh config @Ignas2526 
Could you show me the log @adamwinn thanks!

**Comment by huyphams** [comment]
I checked your console log, the password you entered was wrong
```
[2020/01/25 04:40:53.574289, 1] ssh_packet_userauth_failure:  Access denied for 'password'. Authentication that can continue: publickey,keyboard-interactive
[2020/01/25 04:40:53.574301, 2] ssh_packet_userauth_failure:  Access denied for 'password'. Authentication that can continue: publickey,keyboard-interactive
```
Then the SSH agent send TablePlus a 
```
[2020/01/25 04:40:53.655490, 3] ssh_packet_userauth_info_request:  1 keyboard-interactive prompts
```

This is the popup as "Password:"

TablePlus could not know the content or purpose of the keyboard-interactive prompts, all it does is forward it to user, get the input, and give it back to the agent.

Solution:
Please make sure that you entered the correct password in the form (not on the keyboard-interactive prompts).

**Comment by FAESCH** [comment]
I am having the same issue #1775 

To follow your instructions I deleted my connection and recreated it.
Entered all passwords in the form correctly and hit "Test". Connection worked.

When hitting "connect" on then, I get the same error in the log:
[2020/06/04 08:25:59.620397, 1] ssh_packet_userauth_failure:  Access denied for 'password'. Authentication that can continue: publickey,keyboard-interactive
[2020/06/04 08:25:59.620404, 2] ssh_packet_userauth_failure:  Access denied for 'password'. Authentication that can continue: publickey,keyboard-interactive

**Comment by opensourceame** [comment]
I have this issue with multiple connections. Passwords are _not_ being saved to my keychain.

**Comment by lenuse** [comment]
I also had the same problem, the ssh connection can't store the password and needs to be re-entered every time I connect. Version 4.8.2 (436)

**Comment by nickhammond** [comment]
I'm seeing this on mobile with the same details and key information that I'm utilizing on desktop. Then it eventually says "Invalid private host key" in the mobile app.

**Comment by huyphams** [comment]
Hi @nickhammond did you import the private key to the iOS version?

**Comment by nickhammond** [comment]
@huyphams Yes, going to try it again today/tomorrow but I sort of gave up after a bunch of tries. It used to work just fine on mobile.

**Comment by huyphams** [comment]
Which version of TablePlus are you using @nickhammond ?

**Comment by nickhammond** [comment]
@huyphams It's working now! I noticed there was a new release 4 days ago that mentions updating the SSH driver so I'd guess that fixed it.

**Comment by chrisgo** [comment]
@CalumTowers did you ever figure it out on iOS.  On MacOS, everything is fine but I am importing private keys (both RSA and also another one generated on iPad by Shelly) and still cannot authenticate.  The server has MFA on so it needs a private key and OTP (keyboard-interactive) but I cannot get it to work on iOS.  I was trying to just create the new connection on iOS. 
 I even attempted to use the export from MacOS then import into iOS but also does not work.

**Comment by loofk** [comment]
On macOS 15.1, using SSH Config to connect to Connection will still prompt for SSH password
<img width="1738" alt="image" src="https://github.com/user-attachments/assets/1595c3ef-ce5c-4cba-bfd7-0d322732d000">
